packages/language/src/validation/compiler/IBM1352IE-declare-item-scan-repetition.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { diagnosticFromCode } from "../../language-server/types";
+import {
+  ComputationDataAttribute,
+  DeclaredItem,
+  SyntaxKind,
+  ScanMode,
+  scanModeValues,
+} from "../../syntax-tree/ast";
+import { PLICodes } from "../pli-codes";
+import { ValidationAcceptor } from "../validator";
+
+// %dcl A(4) FIXED INTERNAL SCAN RESCAN; // throws warning IBM3252I with RESCAN.
+// dcl A FIXED SCAN RESCAN; // throws generic IBM1352I with SCAN.
+
+const scanAttributes = (node: DeclaredItem) =>
+  node.attributes
+    .filter((attr) => attr.kind === SyntaxKind.ComputationDataAttribute)
+    .map((attr) => attr as ComputationDataAttribute)
+    .filter((attr) => typeof attr.type === "string" && attr.type !== null)
+    .filter((attr) => scanModeValues.includes(attr.type! as ScanMode));
```

Comment:
Suggestion: I've just noticed that this results in 4 separate iterations on the `attributes` array. Can we merge this into one and just use `as ComputationDataAttribute[]` at the end?